test/literal_test.cpp

```diff
@@ -49,6 +49,20 @@ TEST_CASE(literal_test)
     EXPECT(l4.empty());
 }
 
+TEST_CASE(literal_nstd_shape)
+{
+    migraphx::shape nstd_shape{migraphx::shape::float_type, {1, 3, 2, 2}, {12, 1, 6, 3}};
+    std::vector<float> nstd_data(12);
+    std::iota(nstd_data.begin(), nstd_data.end(), 0);
+
+    migraphx::shape std_shape{migraphx::shape::float_type, {1, 3, 2, 2}};
+    std::vector<float> std_data = {0, 3, 6, 9, 1, 4, 7, 10, 2, 5, 8, 11};
```

Comment:
I think you want to modify this data a bit. 
1 should come on index 3 (starting from 0th index)

Reply:
Here's my work for the striding:
input buffer: [0, 1,  2, 3, 4, 5, 6, 7, 8, 9, 10, 11]
lengths: [1, 3, 2, 2]
strides: [12, 1, 6, 3]

[0, 0, 0, 0] dot [12, 1, 6, 3] = 0 buffer index
[0, 0, 0, 1] dot [12, 1, 6, 3] = 3 buffer index
[0, 0, 1, 0] dot [12, 1, 6, 3] = 6 buffer index
[0, 0, 1, 1] dot [12, 1, 6, 3] = 9 buffer index
[0, 1, 0, 0] dot [12, 1, 6, 3] = 1 buffer index
etc...

The buffer index matches with the data so it's the same.